src/coprocessor/dag/rpn_expr/impl_cast.rs

```diff
@@ -115,6 +173,125 @@ pub fn cast_int_as_decimal(
     }
 }
 
+macro_rules! cast_as_integer {
+    ($ty:ty, $as_int_fn:ident, $as_uint_fn:ident, $as_int_conv:ident, $as_uint_conv:ident) => {
+        cast_as_integer!(
+            _inner,
+            $ty,
+            $as_int_fn,
+            $as_uint_fn,
+            $as_int_conv,
+            $as_uint_conv,
+            val
+        );
+    };
+    ($ty:ty, $as_int_fn:ident, $as_uint_fn:ident, $as_int_conv:ident, $as_uint_conv:ident, $extra:expr) => {
+        cast_as_integer!(
+            _inner,
+            $ty,
+            $as_int_fn,
+            $as_uint_fn,
+            $as_int_conv,
+            $as_uint_conv,
+            $extra
+        );
+    };
+    (_inner, $ty:ty, $as_int_fn:ident, $as_uint_fn:ident, $as_int_conv:ident, $as_uint_conv:ident, $expr:expr) => {
+        #[rpn_fn(capture = [ctx, extra])]
+        #[inline]
+        pub fn $as_int_fn(
```

Comment:
how about the duplicate code when other types are going to be implemented? (e.g. cast_xxx_as_real)

Reply:
I think the ultimate solution is the procedural macro (at least current macro cannot apply to cast_xxx_as_real). 🤔